api/registry/registry_integration.go

```diff
@@ -143,10 +154,22 @@ func (d *idm) Mount(
 		"opts":       opts}
 	ctx.WithFields(fields).Debug("mounting volume")
 
+	ctx = ctx.Join(d.ctx)
+
 	mp, vol, err := d.IntegrationDriver.Mount(
-		ctx.Join(d.ctx), volumeID, volumeName, opts)
+		ctx, volumeID, volumeName, opts)
 	if err != nil {
-		return "", nil, err
+		for x := 0; x < d.retryCount; x++ {
+			mp, vol, err = d.IntegrationDriver.Mount(
+				ctx, volumeID, volumeName, opts)
+			if err == nil {
+				break
+			}
+			time.Sleep(d.retryWait)
```

Comment:
My only comment is whether this `Sleep()` should instead be before the  call to `Mount` inside this `for` loop. That way there will be delay between the initial mount attempt and the first retry. As written, if the retry wait was 1, there would be no delay between the initial attempt and the one and only retry.